src/command_manager.cpp

```diff
@@ -158,81 +158,56 @@ bool CommandManager::stick_deviated(MuxChannel channel)
   uint32_t now = RF_.board_.clock_millis();
 
   // if we are still in the lag time, return true
-  if (now  < rc_stick_override_[channel].last_override_time + RF_.params_.get_param_int(PARAM_OVERRIDE_LAG_TIME))
+  if (now  < channel_override_[channel].last_override_time + RF_.params_.get_param_int(PARAM_OVERRIDE_LAG_TIME))
   {
     return true;
   }
   else
   {
-    if (fabsf(RF_.rc_.stick(rc_stick_override_[channel].rc_channel))
+    if (fabsf(RF_.rc_.stick(channel_override_[channel].rc_channel))
         > RF_.params_.get_param_float(PARAM_RC_OVERRIDE_DEVIATION))
     {
-      rc_stick_override_[channel].last_override_time = now;
+      channel_override_[channel].last_override_time = now;
       return true;
     }
     return false;
   }
 }
-
-bool CommandManager::do_roll_pitch_yaw_muxing(MuxChannel channel)
+uint16_t CommandManager::determine_override_status()
 {
-  bool override_this_channel = false;
-  //Check if the override switch exists and is triggered, or if the sticks have deviated enough to trigger an override
-  if ((RF_.rc_.switch_mapped(RC::SWITCH_ATT_OVERRIDE) && RF_.rc_.switch_on(RC::SWITCH_ATT_OVERRIDE))
-      || stick_deviated(channel))
-  {
-    override_this_channel = true;
-  }
-  else // Otherwise only have RC override if the offboard channel is inactive
+  uint16_t rc_override{OVERRIDE_NO_OVERRIDE};
+  if(RF_.rc_.switch_mapped(RC::SWITCH_ATT_OVERRIDE) && RF_.rc_.switch_on(RC::SWITCH_ATT_OVERRIDE))
+    rc_override |= OVERRIDE_ATT_SWITCH;
+  if(RF_.rc_.switch_mapped(RC::SWITCH_THROTTLE_OVERRIDE) && RF_.rc_.switch_on(RC::SWITCH_THROTTLE_OVERRIDE))
+    rc_override |= OVERRIDE_THR_SWITCH;
+  for(uint8_t channel{0}; channel < MUX_F; channel++)
   {
-    if (muxes[channel].onboard->active)
-    {
-      override_this_channel = false;
-    }
-    else
-    {
-      override_this_channel = true;
-    }
+    if(stick_deviated(static_cast<MuxChannel>(channel)))
+      rc_override |= channel_override_[channel].stick_override_reason;
+    if(!(muxes[channel].onboard->active))
+      rc_override |= channel_override_[channel].offboard_inactive_override_reason;
   }
-  // set the combined channel output depending on whether RC is overriding for this channel or not
-  *muxes[channel].combined = override_this_channel ? *muxes[channel].rc : *muxes[channel].onboard;
-  return override_this_channel;
+  if(!(muxes[MUX_F].onboard->active)) // The throttle has unique override behavior
+    rc_override |= OVERRIDE_OFFBOARD_T_INACTIVE;
+  if(RF_.params_.get_param_int(PARAM_RC_OVERRIDE_TAKE_MIN_THROTTLE))
```

Comment:
I know that the logic is relatively obvious, but I got lost.  A comment would be nice.
also, let's put braces on this `if` statement